umm-spec-lib/src/cmr/umm_spec/test/expected_conversion_util.clj

```diff
@@ -207,7 +207,13 @@
   (let [updated-doi (util/remove-nil-keys
                      (dissoc doi :Authority :MissingReason :Explanation))]
     (if (seq updated-doi)
-      (cmn/map->DoiDoiType updated-doi)
+      (if (:PreviousVersion updated-doi)
+        (-> updated-doi
+            (update-in [:PreviousVersion :Published] #(when % (str %)))
+            (update-in [:PreviousVersion] cmn/map->PreviousVersionType)
+            ;;(update-in [:PreviousVersion] util/remove-nil-keys)
```

Comment:
you want nils?

Reply:
Yes. When convert to PreviousVersionType for umm->dif10->umm,  it will add nils too.  In reality, when you translate, these nils are all removed. I tested it.  But when they show types, they will show these nils at generate-and-parse.

Reply:
should the comment be removed?